rmw_fastrtps_cpp/src/rmw_node.cpp

```diff
@@ -168,14 +168,25 @@ get_security_file_paths(
 
   const char * file_prefix = "file://";
 
-  std::string tmpstr;
+  char * file_path = nullptr;
   for (size_t i = 0; i < num_files; i++) {
-    tmpstr = std::string(rcutils_join_path(node_secure_root, file_names[i]));
-    if (!rcutils_is_readable(tmpstr.c_str())) {
-      return false;
+    file_path = rcutils_join_path(node_secure_root, file_names[i]);
+    if (file_path) {
+      if (!rcutils_is_readable(std::string(file_path).c_str())) {
```

Comment:
There's no real reason to wrap this in a `std::string`, and then get the `c_str` out of it; `file_path` is already a C-style string.  Just do `if (!rcutils_is_readable(file_path)) {`

Reply:
it makes sense, I initially kept it as the original :) and now it's revised